doc/Paypal_Payments.md

```diff
@@ -0,0 +1,159 @@
+# Documentation of PayPal payments
+
+## Steps of a successful donation payment
+
+1. The user submits the donation form
+2. The Fundraising Application stores donation data in the database,
+   generating an ID for the donation. The payment state of the donation is
+   "incomplete"
+3. The Fundraising Application redirects the user to PayPal page, with the
+   donation ID, access token, confirmation and notification routes (see
+   below) as extra URL parameters
+4. The User submits the PayPal form
+5. PayPal redirects the user to success page of the Fundraising
+   Application (passed to PayPal in step 3), passing back the donation ID
+   and access token as URL parameters
+6. PayPal sends a separate HTTP request called **"IPN"** ([Instant Payment
+   Notification][1]
+   to the "PayPal Notification" route (passed to PayPal in step 3) of the
+   Fundraising Application. The HTTP request contains payment metadata
+   (amount, currency, date, account information) and the donation ID.
+7. The Fundraising Application sends the received data back to PayPal to
+   verify that it originated from PayPal.
+8. The Fundraising Application validates the data and marks the payment
+   state donation as "booked".
+
+
+## Metadata for PayPal
+
+When redirecting the user to PayPal, we need to specify the following
+data:
+
+* Amount
+* Currency (Euro)
+* Payment type (one-time payment or subscription)
+* Payment description (e.g. "Ihre Spende für Wikimedia")
+* Receiver ID (email address of WMDE account, `account-address` in our
+  configuration)
+* Donation ID (The [IPN processing][1] will later use the donation ID to
+	associate a PayPal notification to a donation).
+* Access and update tokens for the donation (access token for the user to
+	access the confirmation page after successful payment, update token
+	for PayPal as a small protection against unauthorized access from
+	stray PayPal processes).
+* Return URL (confirmation page)
+* Notification URL (where to send the [IPN][1])
+* Cancel URL (Route to load when the user hits "Cancel")
+
+The class
+`WMDE\Fundraising\PaymentContext\Domain\PaymentUrlGenerator\PayPal`
+creates the URL parameters, getting its configuration from the
+`PayPalConfig` class, which gets initialized with parameters from the
+application configuration file (section `paypal-donation`).
+
+When the URL for receiving notifications changes, we need to adapt it in
+the configuration file.
+
+## IPN Processing
+
+PayPal sends all notifications (IPNs ([Instant Payment Notification][1])
+notification route, but we only process two types of messages:
```

Comment:
"PayPal sends all notifications (IPNs ([Instant Payment Notification][1])
notification route,"
is there some part of the sentence missing here? 
"to the notification route"?